### PR TITLE
Optimize queries

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/fyle/utils.py [hunk 1]
```diff
@@ -172,9 +172,10 @@ def sync_employees(self):
                 }
             })
 
-        employee_attributes = ExpenseAttribute.bulk_upsert_expense_attributes(employee_attributes, self.workspace_id)
+        ExpenseAttribute.bulk_create_or_update_expense_attributes(
+            employee_attributes, 'EMPLOYEE', self.workspace_id)
```
Comment (ashwin1111) [inline apps/fyle/utils.py:176]: We will have to pass `True` for employees

File: apps/mappings/tasks.py [hunk 2]
```diff
@@ -88,37 +88,11 @@ def auto_create_category_mappings(workspace_id):
...
-                detail = {
-                    'source_value': category.value,
-                    'destination_value': category.value
-                }
-                logger.error(
-                    'Error while creating categories auto mapping workspace_id - %s %s',
-                    workspace_id, {'payload': detail}
-                )
-                raise ExpenseAttribute.DoesNotExist
+        category_mappings = Mapping.bulk_create_mappings(fyle_categories, 'CATEGORY', 'ACCOUNT', workspace_id)
```
Comment (ashwin1111) [inline apps/mappings/tasks.py:95]: no use in storing them in a variable

File: apps/mappings/tasks.py [hunk 3]
```diff
@@ -88,37 +88,11 @@ def auto_create_category_mappings(workspace_id):
...
 
         return category_mappings
```
Comment (ashwin1111) [inline apps/mappings/tasks.py:97]: not needed, can return []

File: apps/xero/utils.py [hunk 4]
```diff
@@ -51,7 +50,7 @@ def get_or_create_contact(self, contact_name: str, email: str = None, create: bo
         self.connection.set_tenant_id(tenant_mapping.tenant_id)
 
         contact_name = contact_name.replace('#', '%23')  # Replace '#' with %23
-        contact_name = contact_name.replace('"', '') # remove double quotes from merchant name
+        contact_name = contact_name.replace('"', '')  # remove double quotes from merchant name
```
Comment (ashwin1111) [inline apps/xero/utils.py:53]: why?

File: apps/xero/utils.py [hunk 5]
```diff
@@ -90,8 +89,8 @@ def sync_tenants(self):
                 'destination_id': tenant['tenantId']
             })
 
-        tenant_attributes = DestinationAttribute.bulk_upsert_destination_attributes(
-            tenant_attributes, self.workspace_id)
+        tenant_attributes = DestinationAttribute.bulk_create_or_update_destination_attributes(
+            tenant_attributes, 'TENANT', self.workspace_id)
```
Comment (ashwin1111) [inline apps/xero/utils.py:93]: `True`

File: apps/xero/utils.py [hunk 6]
```diff
@@ -151,7 +156,7 @@ def sync_contacts(self):
 
         for contact in contacts:
             detail = {
-                'email': contact['EmailAddress'] if 'EmailAddress' in contact else None
+                'email': contact['EmailAddress'] if ('EmailAddress' in contact) else None
```
Comment (ashwin1111) [inline apps/xero/utils.py:159]: why is this needed?

Comment (NileshPant1999) [reply apps/xero/utils.py:159]: i think pycharm is automatically inserting it will remove it

**Review by ashwin1111 — APPROVED** [review]

**Review by ashwin1111 — APPROVED** [review]

File: apps/fyle/views.py [hunk 7]
```diff
@@ -420,7 +420,7 @@ def post(self, request, *args, **kwargs):
             fyle_connector = FyleConnector(fyle_credentials.refresh_token, kwargs['workspace_id'])
 
             project_attributes = fyle_connector.sync_projects(active_only=active_only)
-
+
```
Comment (Sravanksk) [inline apps/fyle/views.py:423]: remove this extra tab space

**Review by Sravanksk — APPROVED** [review]
